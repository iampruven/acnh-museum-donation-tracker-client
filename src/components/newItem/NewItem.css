.centered{
    display:flex;
    justify-content: center;
}

.backdrop{
    background-color: white;
    border-radius: 20px;
    border:4px gold solid;
    width: 23%;
    left: 36.5%;
}
.column{
    flex-direction: column;
}

.btn{
    border-radius: 5px;
    background-color:cadetblue;
    margin-bottom:10px;
    margin-top:10px;
    color: white;
    padding: 5px;
    font-size: 25px;
}
.alert{
    color:red;
}

.list-height{
    height:120px;
}
.name-select li{
    cursor: pointer;
}

.name-select li:hover, .name-select li:focus, .name-select li:active {
    background: darkseagreen;
}
input{
    margin: 10px;
    border: 2px solid seagreen;
    border-radius: 4px;
    padding: 12px 20px;
}

label{
    margin: 10px;
}


