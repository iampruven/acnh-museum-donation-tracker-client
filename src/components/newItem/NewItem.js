import React from "react";
import config from '../../config'; 

export default class NewItem extends React.Component {

    state ={
        name:{value:''},
        date:{value:''}
    };
    handleAddItem(name,date){
        fetch(`${config.API_ENDPOINT}/items`,{
            method: 'POST',
            header:{
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                "name":name.value,
                "date": date.value
            })
        })
        .then(res=>{
            if(!res.ok){
                throw new Error(res.statusText)
            }
            return res.json()
        })
        .then(data=>console.log(data))
        .catch(err=>err.message)
      }
    updateName(name){
        this.setState({name:{value:name}})
    }
    updateDate(date){
        this.setState({date:{value:date}})
    }
    validateAddItemForm=(e)=>{
        e.preventDefault()
        console.log(this.state.name.value,this.state.date.value)
        const validName = this.state.name.value;
        const validDate = this.state.date.value;
        
        if(validDate && validName){
            this.handleAddItem(this.state.name, this.state.date)
            this.props.history.push('/')
        }
    }
  render() {
    return (
      <form onSubmit={this.validateAddItemForm}>
        <fieldset>
          <label htmlFor="item">Item:</label>
          <input
            name="item-name"
            id="item"
            type="text"
            placeholder="e.g. tiger butterfly"
            required
            value={this.state.name.value}
            onChange={(e)=>this.updateName(e.target.value)}
            aria-required="true"
            aria-label="Input the item you donated."
          />
          <label htmlFor="date">Date donated:</label>
          <input
            id="date"
            name="date-donated"
            type="date"
            placeholder="03/20/2020"
            required
            value={this.state.date.value}
            onChange={(e)=>this.updateDate(e.target.value)}
            aria-required="true"
            aria-label="Input the data you donated the item to the museum."
          />
          <button type="submit">Submit</button>
        </fieldset>
      </form>
    );
  }
}
