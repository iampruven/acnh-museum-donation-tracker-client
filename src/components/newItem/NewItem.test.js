import React from 'react';
import ReactDOM from 'react-dom';
import NewItem from './NewItem';
import renderer from 'react-test-renderer';

//smoke test
describe('renders without crashing',()=>{
    it('it renders without crashing',()=>{
        const div = document.createElement('div');

        ReactDOM.render(<NewItem />,div);
        ReactDOM.unmountComponentAtNode(div);
    })


//snapshot test
    it('renders the UI as expected', ()=>{
        const tree = renderer
            .create(<NewItem />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })



})