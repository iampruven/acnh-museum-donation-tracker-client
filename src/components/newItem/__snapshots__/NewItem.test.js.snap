// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders the UI as expected 1`] = `
<section
  className="backdrop"
>
  <h1
    className="centered"
  >
    Add New Item!
  </h1>
  <form
    className="column centered"
    onSubmit={[Function]}
  >
    <label
      htmlFor="item"
    >
      Item:
    </label>
    <input
      aria-label="Input the item you donated."
      aria-required="true"
      id="item"
      name="item-name"
      onChange={[Function]}
      placeholder="e.g. tiger butterfly"
      required={true}
      type="text"
      value=""
    />
    <label
      htmlFor="date"
    >
      Date donated:
    </label>
    <input
      aria-label="Input the data you donated the item to the museum."
      aria-required="true"
      id="date"
      name="date-donated"
      onChange={[Function]}
      placeholder="03/20/2020"
      required={true}
      type="date"
      value=""
    />
    <button
      className="btn"
      type="submit"
    >
      Submit
    </button>
  </form>
  <div
    className="alert"
  />
</section>
`;
